pandas/tests/groupby/test_grouping.py

```diff
@@ -202,6 +202,20 @@ def test_grouper_creation_bug(self):
         expected = s.groupby(level='one').sum()
         assert_series_equal(result, expected)
 
+    def test_grouper_func_returning_tuples(self):
+        # GH 22257
+        df = pd.DataFrame({'X': ['A', 'B', 'A', 'B'], 'Y': [1, 4, 3, 2]})
+
+        def func(index):
+            return (df['X'].loc[index], df['Y'].loc[index] % 2)
+
+        gb = df.groupby(by=func, sort=False)
+
+        name, expected = list(gb)[0]
+        result = gb.get_group(name)
```

Comment:
Can you be more explicit about what should be returned here? I think retrieving the element from the GroupBy object and subsequently passing that to `get_group` could potentially hide bugs in the future